pkg/resources/update_job.go

```diff
@@ -19,10 +20,11 @@ const UpdateJobName = "mattermost-update-check"
 func (r *ResourceHelper) LaunchMattermostUpdateJob(
 	jobNamespace string,
 	baseDeployment *appsv1.Deployment,
+	reqLogger logr.Logger,
 ) error {
 	job := PrepareMattermostJobTemplate(UpdateJobName, jobNamespace, baseDeployment)
 
-	err := r.client.Create(context.TODO(), job)
+	err := r.Create(job, baseDeployment, reqLogger)
```

Comment:
This will actually attempt to create a deployment with owner reference to the job. This is perhaps a bit unintuitive but for legacy reasons the first argument to the function is actually the object owner and the second is object to create, take a look at [the function definition](https://github.com/mattermost/mattermost-operator/blob/master/pkg/resources/create_resources.go#L47-L61).

Also we would probably like, the `Mattermost` custom resource to be the owner, so we might need to pass it to the `LaunchMattermostUpdateJob` function.

Reply:
Thanks so much @Szymongib

Reply:
@imskr It actually looks like build issue caught early by the linter. Have you tried running unit tests locally? Try `make unittest`?

Reply:
@imskr yeah it looks like you changed the `Reconcile` method signature: https://github.com/mattermost/mattermost-operator/pull/283/files#diff-0abb08611cc6662ee819cd601d4402335cce0426c2e20ecbc30c7edf1ddd78f4R64

This signature shouldn't be modified as it is a very low level part of the operator that fetches a clusterinstallation custom resource based on the provided context. https://github.com/mattermost/mattermost-operator/blob/master/controllers/mattermost/clusterinstallation/controller.go#L67

It may be a little confusing, but we have two main controllers here for different versions of a mattermost custom resource. The one seen here is the original one and deals with `mattermostv1alpha1.ClusterInstallation` and not the `v1beta1.Mattermost` resource. As such, we shouldn't really pass that in anywhere.